### PR TITLE
Update settings window

### LINKED ISSUE #500: Suggestion: option to set the number of Metatile in the porymap.project.cfg
Currently, changing the number of metablocks can only be done in the source code of Porymap. But in projects where you either want to split metablocks differently (primary and secondary) or are ready to extend them, it would be better if you could set them in the config of Poymap. Especially when several people are working on the project, this would be much easier to control. 

It means this

https://github.com/huderlem/porymap/blob/master/include/core/block.h#L17

### LINKED ISSUE #558: Can paint metatiles on the collision tab at opacity 0
If the opacity slider on the Collision tab is set to the leftmost position/0 (should that even be allowed?) then painting on the map paints metatiles instead of collision/elevation.

### LINKED ISSUE #564: Expose symbol and constant names
Porymap assumes a handful of explicit symbol names (e.g. `gInitialMovementTypeFacingDirections`) and define names/patterns (e.g. `OBJECT_EVENT_TEMPLATES_COUNT` or `\\bOBJ_EVENT_GFX_`). These should be editable in a similar fashion to the project filepaths.

### LINKED ISSUE #565: Porymap can crash closing the settings window
- Open `Options -> Project Settings...` window
- Modify a setting
- Click the window's X to close it (not the `Cancel` button)
- `Yes` to save changes
- `Yes` to reload project
- `OK` to unsaved changes warning. Porymap crashes.

This is just on the master branch, not any releases.

### LINKED ISSUE #566: Porymap can crash if File -> Reload Project fails
If `File -> Open Project` fails the window is disabled. If `File -> Reload Project` fails it isn't disabled, and interacting with most elements of the UI can crash Porymap.

Present since at least release 5.0.0.

### PR DESCRIPTION
**Pull request by GriffinRichards**
- [x] Divides the Project Settings window into tabs so it doesn't get too long vertically
- [x] Adds support for custom block layouts (i.e. specifying number and arrangement of bits given to metatile ID, collision, and elevation). Closes #500 
- [x] As part of supporting the above, add settings to swap out the collision/elevation image selector. This also allows people to make a selector that uses the full range of collision values, or swap it out for aesthetic reasons.
- [x] Update the collision tab UI, including a zoom slider and spinners that let users select the full range of collision/elevation values, regardless of whether they're represented on the selector.
- [x] Adds settings for custom default event icons and custom pokémon icons (see first comment for explanation of the latter)
- [x] Adds a warning to warp events that alerts users if it's not on a warp behavior. The list of behaviors can be edited in the settings
- [x] Adds settings to override the symbol and macro names that Porymap expects. Closes #564
- [x] Fixes #558 
- [x] Fixes #565  
- [x] Fixes #566  
- [x] Update "Editing Map Collisions manual page
- [x] Update "Project Files" manual page
- [x] Update "Settings and Options" manual page
- [x] Update changelog
- [x] More testing

### CODE REVIEW

**Comment by GriffinRichards** [comment]
Updated with two changes that are somewhat related (but mostly dependent on) this PR:
- Porymap requires a specific format and multiple C files to parse the paths of pokémon icons. Now if this fails (which happens even for a few vanilla icons) Porymap will try to predict the path of the icon. If this also fails, a new setting was added to explicitly specify the icon for any species.
- Stop evaluating C defines that Porymap only needs the name of. This has the benefit of being faster, stops some spurious error logging, and gives some lists of defines like flags a more sensible order (the order they were defined in by the project)

**Review by garakmon — COMMENTED** [review]
One general little thing I noticed is that when you close the settings window without saving the changes, it will keep the edited values instead of the "correct" values.

For example, if I changed the first identifier but closed the window without saving the changes it will still load with my garbage text.

Seems great otherwise, although I haven't had time to test every single setting being updated yet/

File: forms/projectsettingseditor.ui
[the remark this answers is not in the document]
Comment (GriffinRichards) [reply forms/projectsettingseditor.ui:1]: No that's correct, that manual page will be updated to also describe the identifiers.

**Comment by GriffinRichards** [comment]
> One general little thing I noticed is that when you close the settings window without saving the changes, it will keep the edited values instead of the "correct" values.
> 
> For example, if I changed the first identifier but closed the window without saving the changes it will still load with my garbage text.
> 

Could you give a more specific example with a set of steps you took? I can't reproduce this. When I edit a setting, close the window, and reopen the window, none of my unsaved changes persist.

**Pull request merged**